pkg/virt-launcher/virtwrap/manager.go

```diff
@@ -1396,6 +1416,53 @@ func (l *LibvirtDomainManager) GetDomainStats() ([]*stats.DomainStats, error) {
 	return l.virConn.GetDomainStats(statsTypes, flags)
 }
 
+func buildDevicesMetadata(vmi *v1.VirtualMachineInstance, dom cli.VirDomain) error {
+	taggedInterfaces := make(map[string]v1.Interface)
+	devicesMetadata := []cloudinit.DeviceData{}
+
+	// Get all tagged interfaces for lookup
```

Comment:
We have mandatory `name` fields on all devices and they are unique across all devices, should they by default be added as a tag too?

Reply:
That could work. I'm just concerned that it might be surprising for users to find metadata for devices that they didn't tag.
I know that some have a script that configures the devices that have been provided via metadata. (like configuring a vlan or ip)